packages/kit/src/runtime/server/respond.js

```diff
@@ -59,6 +59,65 @@ const allowed_page_methods = new Set(['GET', 'HEAD', 'OPTIONS']);
  * @returns {Promise<Response>}
  */
 export async function respond(request, options, manifest, state) {
+	return handle_request(request, options, manifest, state);
+}
+
+/**
+ * @param {import('types').SSROptions} options
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @param {import('types').SSRState} state
+ * @returns {(info: RequestInit | import('crossws').Peer) => Promise<Partial<import('crossws').Hooks>>}
+ */
+export function resolve_websocket_hooks(options, manifest, state) {
+	return async (info) => {
+		/** @type {Request} */
+		let request;
+
+		// Check if info is a Peer object
+		if ('request' in info) {
+			// @ts-ignore the type UpgradeRequest is equivalent to Request
+			request = info.request;
+		} else {
+			// @ts-ignore although the type is RequestInit, it is almost always a Request object
+			request = info;
+		}
+
+		const hooks = await handle_request(request, options, manifest, state, true);
+
+		if (hooks instanceof Response) {
+			return {};
+		}
+
+		return hooks;
+	};
+}
+
+/**
+ * @overload
+ * @param {Request} request
+ * @param {import('types').SSROptions} options
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @param {import('types').SSRState} state
+ * @returns {Promise<Response>}
+ */
+/**
+ * @overload
+ * @param {Request} request
+ * @param {import('types').SSROptions} options
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @param {import('types').SSRState} state
+ * @param {boolean} upgrade
+ * @returns {Promise<Response | Partial<import('crossws').Hooks>>}
+ */
+/**
+ * @param {Request} request
+ * @param {import('types').SSROptions} options
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @param {import('types').SSRState} state
+ * @param {boolean=} upgrade
+ * @returns {Promise<Response | Partial<import('crossws').Hooks>>}
+ */
```

Comment:
I don't know why this is all so complicated to specify that `upgrade` is optional. I think it can just be written as:
```suggestion
/**
 * @param {Request} request
 * @param {import('types').SSROptions} options
 * @param {import('@sveltejs/kit').SSRManifest} manifest
 * @param {import('types').SSRState} state
 * @param {boolean} [upgrade]
 * @returns {Promise<Response | Partial<import('crossws').Hooks>>}
 */
```

Reply:
I added the overload so that TypeScript knows if we specify the `upgrade` parameter that it might return hooks, otherwise, it always only returns a response.

Reply:
Yes this change was restored back to what you had as it was a much better solution

Reply:
oops! sorry I didn't notice the return type was different in the blocks. still I'm not sure why it needs to be three blocks vs just two. I'll leave a suggestion about that